main.py

```diff
@@ -2303,40 +2309,69 @@ def on_modified_async(self):
                             lambda response: self.handle_response(response, pos))
             else:
                 # TODO: this hides too soon.
-                if self.view.is_popup_visible():
+                if self._visible:
                     self.view.hide_popup()
 
     def handle_response(self, response, point):
         if response is not None:
-            config = config_for_scope(self.view)
-            signatures = response.get("signatures")
-            activeSignature = response.get("activeSignature")
-            debug("got signatures, active is", len(signatures), activeSignature)
-            if len(signatures) > 0 and config:
-                signature = signatures[activeSignature]
-                debug("active signature", signature)
-                formatted = []
-                formatted.append(
-                    "```{}\n{}\n```".format(config.languageId, signature.get('label')))
-                params = signature.get('parameters')
-                if params:
-                    for parameter in params:
-                        paramDocs = parameter.get('documentation')
-                        if paramDocs:
-                            formatted.append("**{}**\n".format(parameter.get('label')))
-                            formatted.append("* *{}*\n".format(paramDocs))
-
-                formatted.append(signature.get('documentation'))
-
-                mdpopups.show_popup(
-                    self.view,
-                    preserve_whitespace("\n".join(formatted)),
-                    css=".mdpopups .lsp_signature { margin: 4px; } .mdpopups p { margin: 0.1rem; }",
-                    md=True,
-                    flags=sublime.HIDE_ON_MOUSE_MOVE_AWAY,
-                    location=point,
-                    wrapper_class="lsp_signature",
-                    max_width=800)
+            self._config = config_for_scope(self.view)
+            self.signatures = response.get("signatures")
+            self.active_signature = response.get("activeSignature", -1)
+            debug("got signatures, active is", len(self.signatures), self.active_signature)
+            if len(self.signatures) > 0 and self._config:
```

Comment:
Updating the popup shouldn't care about `_config`, `_build_popup_content` can do its own checking and still function if `languageId` is not available.